server/netty-server/loom/src/test/scala/sttp/tapir/server/netty/loom/NettyIdServerTest.scala

```diff
@@ -1,35 +1,135 @@
+
+    val interpreter = new NettyIdTestServerInterpreter(eventLoopGroup)
+    val createServerTest = new NettyIdCreateServerTest(backend, interpreter)
+    val sleeper: Sleeper[Id] = (duration: FiniteDuration) => Thread.sleep(duration.toMillis)
+
+    val tests =
+      new AllServerTests(createServerTest, interpreter, backend, staticContent = false, multipart = false)
+        .tests() ++
+        new ServerGracefulShutdownTests(createServerTest, sleeper).tests() ++
+        new ServerWebSocketTests(createServerTest, OxStreams, autoPing = true, failingPipe = true, handlePong = true) {
+          override def functionToPipe[A, B](f: A => B): OxStreams.Pipe[A, B] = ox ?=> in => in.map(f)
+          override def emptyPipe[A, B]: OxStreams.Pipe[A, B] = _ ?=> _ => Source.empty
+        }.tests()
+
+    tests.foreach { t =>
+      if (testNameFilter.forall(filter => t.name.contains(filter))) {
+        implicit val pos: Position = t.pos
+
+        this.test(t.name)(t.f())
+      }
+    }
+  }
+  override protected def afterAll(): Unit = {
+    stopBackend.unsafeRunSync()
+    super.afterAll()
+  }
+}
+
+class NettyIdCreateServerTest(
+    backend: SttpBackend[IO, Fs2Streams[IO] & WebSockets],
+    interpreter: NettyIdTestServerInterpreter
+) extends CreateServerTest[Id, OxStreams & WebSockets, NettyIdServerOptions, IdRoute] {
+
+  private val logger = LoggerFactory.getLogger(getClass.getName)
+
+  override def testServer[I, E, O](
+      e: PublicEndpoint[I, E, O, OxStreams & WebSockets],
+      testNameSuffix: String = "",
+      interceptors: Interceptors = identity
+  )(
+      fn: I => Id[Either[E, O]]
+  )(runTest: (SttpBackend[IO, Fs2Streams[IO] & WebSockets], Uri) => IO[Assertion]): Test = {
+    testServerLogic(e.serverLogic(fn), testNameSuffix, interceptors)(runTest)
+  }
+
+  override def testServerLogic(
+      e: ServerEndpoint[OxStreams & WebSockets, Id],
+      testNameSuffix: String = "",
+      interceptors: Interceptors = identity
+  )(
+      runTest: (SttpBackend[IO, Fs2Streams[IO] & WebSockets], Uri) => IO[Assertion]
+  ): Test = {
+    testServerLogicWithStop(e, testNameSuffix, interceptors)((_: IO[Unit]) => runTest)
+  }
+
+  override def testServerLogicWithStop(
+      e: ServerEndpoint[OxStreams & WebSockets, Id],
+      testNameSuffix: String = "",
+      interceptors: Interceptors = identity,
+      gracefulShutdownTimeout: Option[FiniteDuration] = None
+  )(
+      runTest: IO[Unit] => (SttpBackend[IO, Fs2Streams[IO] & WebSockets], Uri) => IO[Assertion]
+  ): Test = {
+    Test(
+      e.showDetail + (if (testNameSuffix == "") "" else " " + testNameSuffix)
+    ) {
+      supervised {
+        val binding = interpreter.scopedServerWithInterceptorsStop(e, interceptors, gracefulShutdownTimeout)
+        val assertion: Assertion =
+          runTest(IO.blocking(binding.stop()))(backend, uri"http://localhost:${binding.port}").guarantee(IO(logger.info(s"Test completed on port ${binding.port}"))).unsafeRunSync()
```

Comment:
We want to run tests synchronously, but our universal `Test` class represents test execution as a `=> Future[Assertion]`, hence this workaround.